### DISCUSSION TITLE
Keeping up to date with the repository

### DISCUSSION THREAD

**Discussion by GalexY727** [Patribots4738/Crescendo2024#63] [Q&A] [answered]
We currently have a discord bot that puts a message for every single comment, commit, PR, issue, discussion, etc all into one channel. This is great and all, but when you realize that we make over 200 contributions per day in this repository it gets a little difficult to manage it all. It would be nice to have a set structured way to notify people of when they are requested to review a PR since our time constraint requires us to be quick on our toes. What do you recommend for you to notify people when they are needed on the repository?

My first thought would be to have a #pull-requests channel on Discord and have a set format for students to follow when a PR is made and requires a short description of the pull request to be made in the channel along with the link to the PR and ping that goes to the requested reviewer to notify them that they are needed. This doesn't resolve the issue of someone getting replied to in the discussions, but that's okay for now. If we want to mention someone about a discussion post we could just ping them and link the post, but I haven't fully fleshed that idea out yet. What are your thoughts?

**Comment by GalexY727** [comment]
To test ping system ima just casually go @derp00monsta @Jacob1010-h @RudyG252 @Oliver-Cushman @rin-star 
Would you say that the ping I just made to you made you look at this discussion or did it make no difference?

**Reply by jacueblol** [reply]
For me, this made no difference because I already have Discord open. Normally, a ping gets my attention cause I also get a notif on Git Hub when I'm asked to review something or something other like that. We could definitely add more channels and make several webhooks on Git Hub for certain things. I always look through the channel, so it doesn't affect me much, but I think a more organized way to view the messages would be nice. If you want, I can get started on that, maybe tomorrow?

Overall I like the ideas of separating PRs but we can also separate many of things...
![image](https://github.com/Patribots4738/Crescendo2024/assets/77470805/0a12ce2a-4ac5-4a24-8bc4-df4343d4fe10)
![image](https://github.com/Patribots4738/Crescendo2024/assets/77470805/7723c725-b5da-4f80-8d56-38c9ab514974)
![image](https://github.com/Patribots4738/Crescendo2024/assets/77470805/12141099-058e-43b9-82c5-5c3e9fb2b831)

**Reply by Oliver-Cushman** [reply]
The ping definitely caught my attention, I just couldn't respond at that moment.

**Reply by tina-hague** [reply]
The ping got my attention because I only get email notifications from GitHub when someone puts my @ but imma change my code.

**Comment by GalexY727** [comment, marked as answer]
So what I am determining from this is that we could have a channel for #issues, #discussions, and #commits. The only thing I would be interested in looking into is getting the attention of a specific person on a specific issue, such as a reviewer for a PR. Oliver says the @ worked for him, and maybe we should get the other programmers to have mobile notifs set up so that they get notified upon being pinged as well.

**Reply by jacueblol** [reply]
I'm marking this as the answer, and I'm going to talk to Thad about this at lunch so that we can get this working.

**Comment by jacueblol** [comment]
Im going to close this issue as it has been resolved, thanks for the help everyone!
